src/baldr/graphreader.cc

```diff
@@ -803,9 +804,16 @@ GraphId GraphReader::GetShortcut(const GraphId& id) {
     uint32_t idx = node->edge_index() + cont_de->opp_index();
     edgeid = {endnode.tileid(), endnode.level(), idx};
     directededge = tile->directededge(edgeid);
+
+    // If this edge is itself not the beginning of a shortcut, but we encountered another shortcut
+    // it means we must have started the traversal outside a shortcuts internal edges
+    if (!directededge->superseded() && shortcut_at_node) {
```

Comment:
i think this works in most cases but it might be overly conservative. i need to review the shortcutbuilder but maybe its possible to have a graph like this


```
graph nodes: a, b, c, d, x
regular edges: .........
shortcut edges: ,,,,,,,,,

                                c
                                .,
                                .,
                                .,
                                .,
                                .,
 ,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,.,
a.............x..........x......b,
                                .,
                                .,
                                .,
                                .,
                                x,
                                .,
                                .,
                                .,
                                .,
                                .,
                                d
```

so here we have a shortcut from `a` to `b` and another one from `c` to `d`. if you take an edge from inside of the latter shortcut and call `GetShortcut` it might find the former when it gets to node `b` and terminate.

now im not sure if this T intersection can happen with shortcuts passing by each other but if it can i think this means we will get a false negative.

Reply:
I could easily be wrong but the way I understand the ShortcutBuilder, a node is only eligible for contraction if it has exactly 2 eligible outbound edges, i.e. it is not a junction of 3 or more eligible roads.

In your example that would mean
- either node `b` cannot be contracted, because it has 3 eligible neighbors, so there might be two shortcuts `a->b`and `b->d` but `b` itself would not be part of one,
- or the outbound edge from `b` towards `a` is a transit edge or under construction, then there would be no shortcut `a->b` and the shortcut `c->d` could safely be resolved by `GetShortcut`.

Either way I'd love if you could confirm these assumptions 😉

Reply:
you've looked at it more recently than i have so i trust your judgement :smile: i mean in any case what we are doing here is better than what we had before so it can only become more accurate.